brave/src/main/java/brave/propagation/CurrentTraceContext.java

```diff
@@ -53,6 +51,7 @@
      */
     public Builder addScopeDecorator(ScopeDecorator scopeDecorator) {
       if (scopeDecorator == null) throw new NullPointerException("scopeDecorator == null");
+      if (scopeDecorator == ScopeDecorator.NOOP) return this;
```

Comment:
we need this because customizers act on builders.. we don't always know if there will be any configuration at all.